### DISCUSSION TITLE
Getting absolute file path

### DISCUSSION THREAD

**Discussion by jiridusil** [cypress-io/cypress#29631] [Questions and Help]
Before running a test I need to update a yaml file on a file system outside of 'cypress' folder.

Example:
Under 'fixtures' folder I have a following directory structure:
-> env1 -> config.yaml
-> env2 -> config.yaml
-> env3 -> config.yaml

Before running a test I need to copy particular **config.yaml** from above to folder 'Users/jiri/source/'.

I want to use `cy.exec('cp source_file target_folder');`

The question is how to get an absolute (full) path of the **config.yaml** file from 'fixtures'. Thank you so much in advance.

**Comment by portyu9** [comment]
Hi @jiridusil 

For the Cypress version from this thread, you could use a path relative to the Cypress project root, e.g. `cypress/fixtures/env1/config.yaml`.

For current Cypress, however, `cy.exec()` has been removed. The supported approach is to perform the copy in a `cy.task()` using Node's filesystem APIs:

```js
const fs = require('node:fs')
const path = require('node:path')

e2e: {
  setupNodeEvents(on, config) {
    const fixtures = path.resolve(
      config.projectRoot,
      config.fixturesFolder
    )

    on('task', {
      copyConfig(environment) {
        const source = path.join(
          fixtures,
          environment,
          'config.yaml'
        )

        fs.copyFileSync(
          source,
          '/Users/jiri/source/config.yaml'
        )

        return null
      },
    })
  },
}
```

Then from the test:

```js
cy.task('copyConfig', 'env1')
```

This avoids shell/path quoting issues and gives you the resolved filesystem location directly in Node.

Please mark this answer as accepted if it helped, thank you.
